envoy/http/filter.h

```diff
@@ -174,11 +174,20 @@ enum class FilterTrailersStatus {
 
 /**
  * Return codes for encode metadata filter invocations. Metadata currently can not stop filter
- * iteration.
+ * iteration except in the case of local replies.
  */
 enum class FilterMetadataStatus {
-  // Continue filter chain iteration.
+  // Continue filter chain iteration for metadata only. Does not unblock returns of StopIteration*
```

Comment:
Is this the behavior for any other filter chain continue? I'd prefer to not add more complexity to the filter chain if we can avoid it.

Reply:
Yeah I agree this is quite confusing. Renaming at minimum sounds fine to me if the alternative is too risky (agreed it probably is)

Reply:
I'd suggested a follow-up, I'd also be fine with you just shipping it out as a separate PR but I'd definitely like to keep the high risk changing-the-filter-chain PR separate from the renaming-a-field-in-many-places-PR :-)
/wait

Reply:
Great -- I reverted the rename, and will move it to a followup PR.